### PR TITLE
Fix test client setup and result interface tests

### PR DESCRIPTION
**Pull request by NRHelmi**
This PR:
* fixes the test client configuration
* logs used engine, database and custom headers if they ever exist
* fixes the failing result interface tests

### CODE REVIEW

**Comment by larf311** [comment]
In the future, please split these into different PRs, one for just the logging / headers changes and then one for the actual test fixes. It makes the PRs smaller and easier to review. And if one has a problem the other could still get merged.

**Review by larf311 — COMMENTED** [review]
Do we log the environment and the account name anywhere?

File: rai/main_test.go [hunk 1]
```diff
@@ -91,17 +93,21 @@ func (h headerRoundTrip) RoundTrip(r *http.Request) (*http.Response, error) {
...
-	clientSecret := os.Getenv("CLIENT_SECRET")
-	clientCredentialsUrl := os.Getenv("CLIENT_CREDENTIALS_URL")
+		clientId := os.Getenv("CLIENT_ID")
+		clientSecret := os.Getenv("CLIENT_SECRET")
+		clientCredentialsUrl := os.Getenv("CLIENT_CREDENTIALS_URL")
 
-	placeHolderConfig := `
+		configFormat := `
 		[default]
 		host=azure.relationalai.com
```
[the remark this answers is not in the document]
Comment (NRHelmi) [reply rai/main_test.go:112]: in SDK repository we run the tests against production, in raicode repository we use configuration inside `~/.rai/config` so whatever environment is setup there we run against it
I think we need to make the host field also configurable in this section

**Comment by NRHelmi** [comment]
> Do we log the environment and the account name anywhere?

We don't, not sure from where we can get this information ?

File: rai/main_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (larf311) [reply rai/main_test.go:112]: The problem with running against production is that customer headers don't work in production.

We should implement a similar action as this one for running the tests so we can pass it in:
https://github.com/RelationalAI/rai-sdk-python/blob/main/.github/actions/test/action.yml
That is outside the scope of this PR though.

@miazamrai was going to do this but perhaps you can go ahead and add it in an separate PR?

[the remark this answers is not in the document]
Comment (NRHelmi) [reply rai/main_test.go:112]: sure will address this in a separate PR

**Review by miazamrai — APPROVED** [review]

File: rai/main_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (NRHelmi) [reply rai/main_test.go]: thanks @miazamrai for addressing the requirement https://github.com/RelationalAI/rai-sdk-go/pull/45

**Review by larf311 — APPROVED** [review]

**Pull request merged**